### PR TITLE
:sparkles: Update event controls.

### PR DESCRIPTION
**Pull request by ParthBaraiya**
- Added method to update filter.

- Removed tree structure to store single events. Use map instead.

- Add comparison for startTime and endTime in CalendarEventData.

### CODE REVIEW

File: lib/src/calendar_event_data.dart [hunk 1]
```diff
@@ -69,7 +69,16 @@ class CalendarEventData<T extends Object?> {
     return other is CalendarEventData<T> &&
         date.compareWithoutTime(other.date) &&
         endDate.compareWithoutTime(other.endDate) &&
-        event == other.event &&
+        ((event == null && other.event == null) ||
+            (event != null && other.event != null && event == other.event)) &&
```
Comment (vatsaltanna) [inline lib/src/calendar_event_data.dart:73]: what is the reason for checking == null as well as != null for the same parameter?

File: lib/src/extensions.dart [hunk 2]
```diff
@@ -110,6 +106,18 @@ extension DateTimeExtensions on DateTime {
         totalMinutes ~/ 60,
         totalMinutes % 60,
       );
+
+  /// Returns [DateTime] without timestamp.
+  DateTime get withoutTime => DateTime(year, month, day);
+
+  /// Compares time of two [DateTime] objects.
+  bool hasSameTimeAs(DateTime other) {
```
Comment (vatsaltanna) [inline lib/src/extensions.dart:114]: there one function already available for the same purpose `isAtSameMomentAs`, any other reason for adding this ?

Comment (ParthBaraiya) [reply lib/src/extensions.dart:114]: We need to check only time in DateTime object regardless of what date is there. `isAtSameMomentAs` does not provide that.

File: lib/src/calendar_event_data.dart (hunk 1 above)
Comment (ParthBaraiya) [reply lib/src/calendar_event_data.dart:73]: Basically event is nullable. So, logic here is if both events are null or both are non null and same objects then we can consider it as same. If not then they are not same object.

**Review by vatsaltanna — APPROVED** [review]

**Pull request merged**